src/Type/MenuItem/Connection/MenuItemConnectionResolver.php

```diff
@@ -0,0 +1,143 @@
+<?php
+namespace WPGraphQL\Type\MenuItem\Connection;
+
+use GraphQL\Type\Definition\ResolveInfo;
+use GraphQLRelay\Connection\ArrayConnection;
+use GraphQLRelay\Relay;
+use WPGraphQL\AppContext;
+use WPGraphQL\Type\PostObject\Connection\PostObjectConnectionResolver;
+
+/**
+ * Class ThemeConnectionResolver
+ *
+ * @package WPGraphQL\Data\Resolvers
+ * @since 0.5.0
+ */
+class MenuItemConnectionResolver extends PostObjectConnectionResolver {
+
+	/**
+	 * This returns the $query_args that should be used when querying for posts in the postObjectConnectionResolver.
+	 * This checks what input $args are part of the query, combines them with various filters, etc and returns an
+	 * array of $query_args to be used in the \WP_Query call
+	 *
+	 * @param mixed       $source  The query source being passed down to the resolver
+	 * @param array       $args    The arguments that were provided to the query
+	 * @param AppContext  $context Object containing app context that gets passed down the resolve tree
+	 * @param ResolveInfo $info    Info about fields passed down the resolve tree
+	 *
+	 * @return array
+	 * @throws \Exception
+	 */
+	public static function get_query_args( $source, array $args, AppContext $context, ResolveInfo $info ) {
+
+		/**
+		 * Determine the menu_slug based on the $source of the query
+		 */
+		if ( $source instanceof \WP_Term ) {
+			$menu_slug = ! empty( $source->slug ) ? $source->slug : null;
+		} elseif ( $source instanceof  \WP_Post ) {
+			$menu_slug = ! empty( $source->menu->slug ) ? $source->menu->slug : null;
+		}
+
+		/**
+		 * If menuSlug was entered as an argument, use it
+		 */
+		if ( ! empty( $args['where']['menuSlug'] ) ) {
+			$menu_slug = sanitize_text_field( $args['where']['menuSlug'] );
+		}
+
+		/**
+		 * If the menuLocation is set in the $args, use it to set the menu slug
+		 */
+		if ( ! empty( $args['where']['menuLocationSlug'] ) ) {
+			$theme_locations = get_nav_menu_locations();
+			$menu = wp_get_nav_menu_object( $theme_locations[ $args['where']['menuLocationSlug'] ] );
+			$menu_slug = ! empty( $menu->slug ) ? $menu->slug : $menu_slug;
+		}
+
+		/**
+		 * If the source of the query is another nav_menu_item, and
+		 * the field is "childItems" set the value of the $menu_item_parent to
+		 */
+		if (
+			$source instanceof \WP_Post &&
+			'nav_menu_item' === get_post_type( $source ) &&
+			'childItems' === $info->fieldName
+		) {
+			$parent_id = $source->ID;
+		} else {
+			$parent_id = 0;
+		}
+
+
+		/**
+		 * If an argument was passed for a specific parentId, use it
+		 */
+		if ( ! empty( $args['where']['parentMenuItemId'] ) ) {
+			$id_parts = Relay::fromGlobalId( $args['where']['parentMenuItemId'] );
+			if ( is_array( $id_parts ) && ! empty( $id_parts['id'] ) ) {
+				$parent_id = $id_parts['id'];
+			}
+		}
+
+		$query_args['post_type'] = 'nav_menu_item';
+		$query_args['posts_per_page'] = 100;
```

Comment:
So this will affect each level, not the menu in it's entirety. 

I have a hard time thinking any individual level of a menu would have more than 100 items, but ya, there should definitely be a filter. 

I actually implemented one in the `get_query_amount` method which is part of the ConnectionResolver class that we're extending here, so I'll adjust to make use of that so that the filter can be applied for menus in the same way it is able to be filtered for other post_types with a filter like so: 

```
add_filter( 'graphql_connection_max_query_amount', function( $amount, $source, $args, \WPGraphQL\AppContext $context, \GraphQL\Type\Definition\ResolveInfo $info ) {

	if ( 'menuItems' === $info->fieldName ) {
		$amount = 1;
	}

	return $amount;

}, 10, 5 );
```